src/lib/operators/import_binary.cpp

```diff
@@ -224,4 +249,75 @@ std::shared_ptr<RunLengthSegment<T>> ImportBinary::_import_run_length_segment(st
   return std::make_shared<RunLengthSegment<T>>(values, null_values, end_positions);
 }
 
+template <typename T>
+std::shared_ptr<FrameOfReferenceSegment<T>> ImportBinary::_import_frame_of_reference_segment(std::ifstream& file,
+                                                                                             ChunkOffset row_count) {
+  const auto attribute_vector_width = _read_value<AttributeVectorWidth>(file);
+  const auto block_count = _read_value<uint32_t>(file);
+  const auto block_minima = pmr_vector<T>(_read_values<T>(file, block_count));
+  const auto size = _read_value<uint32_t>(file);
+  const auto null_values = pmr_vector<bool>(_read_values<bool>(file, size));
+  auto offset_values = _import_offset_value_vector(file, row_count, attribute_vector_width);
+
+  return std::make_shared<FrameOfReferenceSegment<T>>(block_minima, null_values, std::move(offset_values));
+}
+
+template <typename T>
+std::shared_ptr<LZ4Segment<T>> ImportBinary::_import_lz4_segment(std::ifstream& file, ChunkOffset row_count) {
+  const auto num_elements = _read_value<uint32_t>(file);
+  const auto block_count = _read_value<uint32_t>(file);
+
+  uint32_t block_size;
+  uint32_t last_block_size;
+  if (block_count > 1) {
+    block_size = _read_value<uint32_t>(file);
+    last_block_size = _read_value<uint32_t>(file);
+  } else {
+    last_block_size = _read_value<uint32_t>(file);
+    block_size = last_block_size;
+  }
```

Comment:
Seeing how this is handled here - does the special treatment make anything easier or could we just write the same value twice to remove one edge case?

Reply:
doesn't make anything easier. just saves 4 bytes. should we rather go for low complexity than low memory footprint?

Reply:
I'd suggest so, but if you like the current version better, I'm fine with it